### PR TITLE
@types/graphql: is*Type() predicates can take null and undefined

### PR DESCRIPTION
**Pull request by benwaffle**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://graphql.org/graphql-js/type/#isinputtype
- [ ] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by bradzacher — APPROVED** [review]
LGTM

**Review by IvanGoncharov — CHANGES_REQUESTED** [review]
Just change types to `any` so it would match `graphql-js`:
https://github.com/graphql/graphql-js/blob/master/src/type/definition.js#L434

File: types/graphql/type/definition.d.ts [hunk 1]
```diff
@@ -36,35 +36,35 @@ export function isType(type: any): type is GraphQLType;
 
 export function assertType(type: any): GraphQLType;
 
-export function isScalarType(type: GraphQLType): type is GraphQLScalarType;
+export function isScalarType(type: any): type is GraphQLScalarType;
 
 export function assertScalarType(type: GraphQLType): GraphQLScalarType;
```
Comment (IvanGoncharov) [inline types/graphql/type/definition.d.ts]: @benwaffle To keep definitions consistent can you please also change `assert*Type` functions?

Comment (benwaffle) [reply types/graphql/type/definition.d.ts]: updated

**Review by IvanGoncharov — APPROVED** [review]
@benwaffle Thanks, looks good now 👍

**Pull request merged**